lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -804,12 +808,12 @@ WasmBinaryReader::ReadElementSection()
     for (uint32 i = 0; i < count; ++i)
     {
         uint32 index = LEB128(length); // Table id
-        if (index != 0)
+        if (index != 0 || !(m_module->HasTable() || m_module->HasTableImport()))
```

Comment:
It feels weird to allow to set data or elements in a memory/table that hasn't been defined.
In our current implementation, we would throw a linking error when treating the data/element segments.
This is only moving the error to compile time effectively changing from WebAssembly.RuntimeError to WebAssembly.CompileError which is more in line with the spec interpreter.
In the end, it really doesn't matter, but early error should be favorable if we have to read the data anyway